### PR TITLE
Adding Feature of Authentication

### PR DESCRIPTION
**Pull request by JavidSumra**
- [x] *Have you familiarized yourself with the [Contributing Guidelines](https://github.com/jessesquires/.github/blob/master/CONTRIBUTING.md)?*
Issue #37 

## Changes Description
Implemented seamless user authentication features including login, signup, and logout. These enhancements elevate the overall user experience, ensuring secure and user-friendly authentication functionalities.

### CODE REVIEW

**Review by anand-harsh — CHANGES_REQUESTED** [review]
If user is not logged in then he cant access anything in application, just show Login and signup button on page, and after successful login, redirect to main application. Hope you get that. Also for knowing whether user is logged in or not, change name on view profile by name of user or email of user

**Comment by JavidSumra** [comment]
Hey @anand-harsh I will do it by tomorrow and yes you need to add an environment variable named `REACT_APP_API_ENDPOINT` with the value of the backend url in production.

**Comment by JavidSumra** [comment]
Hey @anand-harsh Sir Can You Please Add `Labels` over here?

**Comment by anand-harsh** [comment]
Hi, the auth part is not working for me in my local, can you guide me what things I need to do to?, like env

**Comment by JavidSumra** [comment]
Hey @anand-harsh  you need to add an environment variable named `REACT_APP_API_ENDPOINT` with the value of the backend url in production.

File: client/src/components/Auth/Register.jsx [hunk 1]
```diff
@@ -38,8 +43,58 @@ const Register = () => {
...
+
+  const handleSubmit = async e => {
+    e.preventDefault();
+    try {
+      const res = await fetch(`${API_ENDPOINT}/register`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ email, name, password, image }),
```
Comment (anand-harsh) [inline client/src/components/Auth/Register.jsx:55]: Don't take image for now, just take name, email, password

**Review by anand-harsh — CHANGES_REQUESTED** [review]
The content of other pages are not visible after login, fix it

**Comment by JavidSumra** [comment]
> The content of other pages are not visible after login, fix it

Hey @anand-harsh Can You Please Share Screenshot with Console?

File: client/src/components/Auth/Login.jsx [hunk 2]
```diff
@@ -7,20 +7,77 @@ import {
...
+      });
+
+      const data = await res.json();
+
+      console.log(data);
+      if (data?.success) {
+        localStorage.setItem('isAuth', true, 3600000);
+        localStorage.setItem(
+          'userData',
+          JSON.stringify({ User: data?.user }),
```
Comment (anand-harsh) [inline client/src/components/Auth/Login.jsx:42]: Just add email and createdAt feild in localStorage, dont add whole data

**Review by anand-harsh — CHANGES_REQUESTED** [review]
Do changes and let me know

**Comment by JavidSumra** [comment]
Hey @anand-harsh From the Backend I am Just Getting Data Like Name,email and createdAt for Login

**Comment by anand-harsh** [comment]
> Hey @anand-harsh From the Backend I am Just Getting Data Like Name,email and createdAt for Login

No issues, just dont add all data to localStorage, as its vulnerable

**Comment by JavidSumra** [comment]
Hey @anand-harsh Changes Done!!!

**Comment by anand-harsh** [comment]
Hi @JavidSumra , the pages are still not visible, come to discord

**Review by anand-harsh — APPROVED** [review]
LGTM✨✨

**Pull request merged**